### PR TITLE
Add utility rate forecast example for new stateful cmod

### PR DESCRIPTION
**Pull request by brtietz**
PySAM example for new cmod utility rate forecast

What is best practice with files in docs, modules, and stubs? This work generated new files there, but I wasn't sure if they should be committed. Here's what my untracked files look like:

image

Companion PRs:
SSC: https://github.com/NREL/ssc/pull/868
SAM: https://github.com/NREL/SAM/pull/1144

### CODE REVIEW

**Comment by sjanzou** [comment]
MacOS 12.5.1 Apple silicon results

python UtilityRateForecastExample.py 
urdb_rate_618940545457a35a1c4097ec.json
dict_keys(['ElectricityRates', 'Lifetime', 'Controls', 'Outputs'])
((0.0,), (1.0094e-320,), (5.776622002767455e-275,), (4.005296727656348e-307,), (0.0,), (1.344974619049452e-284,), (4.00530012336606e-307,), (0.0,), (1.344974619049452e-284,), (0.0,), (0.0,), (0.0,))
970.3978152001135
8760.0
1303.3002455446654

File: Examples/UtilityRateForecastExample.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    print(rate_forecast.export().keys())
+    print(rate_forecast.export()["ElectricityRates"]["ur_dc_peaks"])
+    print(rate_forecast.export()["Outputs"]["ur_total_bill"]) # Note that this does not include fixed charges
+    print(rate_forecast.export()["Controls"]["idx"])
+
+    price_series = rate_forecast.export()["Outputs"]["ur_price_series"]
+    print(sum(np.array(load) * np.array(price_series)))
+
+    #rate_forecast.value("grid_power", grid_power[4380:8760]) # Hourly kW
+    #rate_forecast.execute()
```
Comment (dguittet) [inline Examples/UtilityRateForecastExample.py:80]: Why is this commented out?

File: Examples/UtilityRateForecastExample.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import json
+import os
+import requests
+import numpy as np
+import pandas as pd
+
+import PySAM.Utilityrateforecast as utility_rate_forecast
+import PySAM.ResourceTools
+
```
Comment (dguittet) [inline Examples/UtilityRateForecastExample.py:9]: I think it would be really helpful if there was a short intro paragraph explaining what this is doing

**Comment by dguittet** [comment]
@brtietz Ah good questions about files. The RST, C and pyi files should be committed. the binaries shouldn't be and I'm not sure what tmp_stubs is from

**Comment by dguittet** [comment]
Also, it would be great to have a test for this new module.

**Comment by brtietz** [comment]
Error if urdb7.json was unchanged:
image

Correcting the JSON file doesn't affect previous utility rate tests, and allows the new tests to ass

**Review by dguittet — APPROVED** [review]

**Pull request merged**